temporal-sdk/src/main/java/io/temporal/common/converter/EncodedValues.java

```diff
@@ -37,6 +37,7 @@ public EncodedValues(Optional<Payloads> payloads, DataConverter converter) {
 
   public EncodedValues(Object... values) {
     this.values = values;
+    this.converter = DefaultDataConverter.getDefaultInstance();
```

Comment:
This is not going to work as it is going to ignore custom DataConverters set through options.

Reply:
Yes, I didn't see an option of setting it in the ManualCompletionClient and thought that this would be a better tradeoff than changing exceptionToFailure method signature.
Changed to the option that would work with custom converters too.